magenta/reviews/GAN.md

```diff
@@ -0,0 +1,136 @@
+from the pool generated by the generator.
+The discriminator is trained in a supervised manner with two labels -- Real and Generated.
+The generator is trained to maximize the discriminator’s error on generated examples.
+The result is that it learns how to generate examples that approximate the real data distribution.
+With GANs, we approximate it by optimizing an adversarial loss:
+
+<p align="center">
+  <img align="center" src="assets/gan/image07.png"><br>
+</p>
+
+Those familiar with game theory might appreciate the insight that the optimization of this objective is like finding
+a Nash equilibrium of a minimax game between the generator and the discriminator.
+The loss has two components.
+The first one, <img src="assets/gan/image06.png">, makes the discriminator <img src="assets/gan/image01.png"> better at
+discriminating real examples from fake ones, and the second one, <img src="assets/gan/image13.png">, makes the generator
+<img src="assets/gan/image14.png"> generate samples that the discriminator considers real.
+The input, <img src="assets/gan/image02.png">, is a noise vector with distribution <img src="assets/gan/image10.png">.
+This is usually an N-dimensional normal or uniform distribution and serves as input to the network.
+
+Both parts can be independently optimized with gradient based methods.
+First, do an optimization step for the discriminator to maximize <img src="assets/gan/image08.png">, then do
+an optimization step for the generator to make it better at fooling this new discriminator by minimizing
+<img src="assets/gan/image12.png">. If <img src="assets/gan/image03.png"> is very low, which is the case early in training,
+<img src="assets/gan/image12.png"> is very close to zero and the generator cannot learn quickly.
+This can be fixed by maximizing <img src="assets/gan/image11.png"> instead, which converges to the same solution, but
+provides strong gradients when <img src="assets/gan/image03.png"> is low.
+
+Here’s a visual explanation of how the learning happens.
+
+<p align="center">
+ <img src="assets/gan/image09.png"><br>
+ <i>Training a GAN to fit a data distribution.</i>
+</p>
+
+Imagine that the real data is represented by some distribution, drawn as a dotted black line on the figure.
+Real examples then are obtained by sampling from it.
+<img src="assets/gan/image04.png"> is the domain of the model's input.
+Fake examples are obtained by sampling from some distribution on domain <img src="assets/gan/image02.png">.
+The generator maps this <img src="assets/gan/image02.png"> to <img src="assets/gan/image04.png"> which results in fake examples,
+whicha are distributed according to the green line on the figure.
+The dashed blue line is the discriminator and it gives each example from <img src="assets/gan/image04.png"> a probability of
+being real.
+
+Starting with a fixed, random generator and a discriminator (see Figure (a)), we begin by training the discriminator
```

Comment:
"Starting with a fixed, random generator and a discriminator". Please fix this grammar. If I didn't know this work, I would not understand this sentence. What you mean to say is that the models start from a fixed but random initialization. Then say that. Be like Hemingway.